bindings_generator/src/methods.rs

```diff
@@ -3,126 +3,119 @@ use crate::documentation::class_doc_link;
 use crate::rust_safe_name;
 use crate::GeneratorResult;
 
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote};
+
 use std::collections::HashSet;
-use std::fmt;
 use std::io::Write;
 
 fn skip_method(name: &str) -> bool {
-    name == "free" || name == "reference" || name == "unreference"
+    const METHODS: &[&str] = &["free", "reference", "unreference"];
+    METHODS.contains(&name)
 }
 
 pub fn generate_method_table(
     output: &mut impl Write,
+    api: &Api,
     class: &GodotClass,
-    has_underscore: bool,
 ) -> GeneratorResult {
-    writeln!(
-        output,
-        r#"
-#[doc(hidden)]
-#[allow(non_camel_case_types)]
-pub(crate) struct {name}MethodTable {{
-    pub class_constructor: sys::godot_class_constructor,"#,
-        name = class.name
-    )?;
-
-    for method in &class.methods {
-        let MethodName {
-            rust_name: method_name,
-            ..
-        } = method.get_name();
-        if method_name == "free" {
-            continue;
-        }
-        writeln!(
-            output,
-            "    pub {}: *mut sys::godot_method_bind,",
-            method_name
-        )?;
-    }
-    writeln!(
-        output,
-        r#"
-}}
-
-impl {name}MethodTable {{
-    unsafe fn get_mut() -> &'static mut Self {{
-        static mut TABLE: {name}MethodTable = {name}MethodTable {{
-            class_constructor: None,"#,
-        name = class.name
-    )?;
-    for method in &class.methods {
-        let MethodName {
-            rust_name: method_name,
-            ..
-        } = method.get_name();
-        if method_name == "free" {
-            continue;
-        }
-        writeln!(
-            output,
-            "            {}: 0 as *mut sys::godot_method_bind,",
-            method_name,
-        )?;
-    }
+    let has_underscore = api.api_underscore.contains(&class.name);
+
+    let method_table = format_ident!("{}MethodTable", class.name);
+    let _class_name = format_ident!("{}", class.name);
```

Comment:
Why underscore? Unused let-binding?

Reply:
I'd been having an issue with my editor not 'quick fixing' the right thing. :D fixed.